code/game/objects/items/robot/robot_parts.dm

```diff
@@ -417,16 +439,47 @@
 			log_silicon("[key_name(user)] has [aisync ? "enabled" : "disabled"] the AI sync for a cyborg shell at [loc_name(user)]")
 			return TRUE
 		if("set_ai")
-			var/selected_ai = select_active_ai(user, z)
-			if(!in_range(src, user) && loc != user)
-				return
-			if(!selected_ai)
+			if(length(active_ais(check_mind = FALSE, z = z)) <= 0)
```

Comment:
are we getting a negative `length()` somehow? why not just do a `if(!length(thing))`? or just `if(length(thing) == 0)`? the `<` is confusing

Reply:
i just do it because i find it more readable that `!length`

Reply:
that's fine then but i just have no clue why the code would insinuate that length could ever be less than zero